lib/engine/game/g_1824.rb

```diff
@@ -89,14 +211,211 @@ def setup
       end
 
       def ipo_name(entity)
-        return 'Treasury' if @coal_corporations.include?(entity)
+        return 'Treasury' if coal_railways.include?(entity)
 
         'IPO'
       end
 
       def can_par?(corporation, parrer)
         super && !corporation.all_abilities.find { |a| a.type == :no_buy }
       end
+
+      private
+
+      def coal_railways
+        @coal_railways ||= (option_cislethania ? %w[EPP EOD MLB] : %w[EPP EOD MLB SPB])
+          .map { |c| corporation_by_id(c) }
+      end
+
+      def validate_optional_rules(optional_rules)
+        if optional_rules&.include?(:cisleithania)
+          raise GameError 'Cisleithania optional rule is for 2-3 players' if @players.size > 3
+          raise GameError 'Cannot use Cisleithania and Goods Time at the same time' if option_goods_time
+        elsif @players.size < 3
+          raise GameError '2 player count requires to use Cisleithania optional rule'
+        end
+      end
+
+      def mountain_railway_definition(index)
```

Comment:
put all of these definitions in constants not in methods like this

Reply:
Have attempted to fix this.